commerce_coordinator/apps/titan/pipeline.py

```diff
@@ -60,14 +68,39 @@ def run_filter(self, edx_lms_user_id, payment_number=None, **kwargs):  # pylint:
         try:
             payment = api_client.get_payment(
                 edx_lms_user_id=edx_lms_user_id,
-                payment_number=payment_number
             )
         except HTTPError as exc:
             logger.exception("[GetTitanPayment] Payment %s not found for user: %s", payment_number, edx_lms_user_id)
             raise PaymentNotFound from exc
         payment_serializer = PaymentSerializer(data=payment)
         payment_serializer.is_valid(raise_exception=True)
-        return payment_serializer.data
+        payment = payment_serializer.data
+
+        order_uuid = kwargs.get('order_uuid')
+        if order_uuid and payment['order_uuid'] != kwargs['order_uuid']:
+            raise InvalidOrderPayment(
+                f'Requested order_uuid "{order_uuid}" does not match with '
+                f'order_uuid "{payment["order_uuid"]}" in Spree system.'
+            )
+
+        if payment_number and payment['payment_number'] != payment_number:
+            raise PaymentMismatch(
+                f'Requested payment number "{payment_number}" does not match with '
+                f'payment number "{payment["payment_number"]}" in Spree system.'
+            )
+
+        if kwargs.get('validate_payment_processing_state'):
```

Comment:
Would it make sense to explicitly add `kwargs.get('validate_payment_processing_state')` as an optional argument to this pipeline step?

Also, would `validate_payment_in_draft_state` be a better name? (It may not be; I could be misunderstanding.)